autowarefoundation/autoware.universe
# feat(default_ad_api): add localization api

Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## Description

Introduce AD API for routing. This PR is the rviz helper, auto initial pose and AD API part of the diagram.
Here are the changes for each package:

autoware_ad_api_msgs: Add the localization API messages.
autoware_ad_api_specs: Add the localization API definitions.
component_interface_specs; Add the internal localization interface definitions.
default_ad_api: Add the localization API.
default_ad_api_helpers: Add the adaptor for RViz localization topics, and auto initialize node by GNSS.

![block-diagram-initial-pose drawio](https://user-images.githubusercontent.com/43976882/182350009-11922fc7-da2b-4f3f-81ee-33a87bcf09ab.png)

## Related links

https://github.com/autowarefoundation/autoware_launch/pull/72
https://github.com/autowarefoundation/autoware_launch/pull/70
https://github.com/tier4/tier4_autoware_msgs/pull/44
https://github.com/autowarefoundation/autoware.universe/pull/1500


~~https://github.com/autowarefoundation/autoware_launch/pull/71~~

## Tests performed

For planning simulation.
1. Merge all related PRs.
1. Launch planning simulation.
1. Check if the localization succeeds when initialpose from RViz.


For logging simulation.
1. Merge all related PRs.
1. Launch logging simulation (sample map).
1. Check if the localization succeeds automatically when the vehicle is stopped.
1. Disable auto initial pose (https://github.com/autowarefoundation/autoware_launch/pull/71)
   - Check if the localization succeeds when initialpose from RViz when vehicle is stopped.
   - Check if the localization failes when initialpose from RViz when vehicle is not stopped.


## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
